arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/RaiseAccumulate.kt

```diff
@@ -476,6 +478,45 @@ public inline fun <Error, A, B> Raise<NonEmptyList<Error>>.mapOrAccumulate(
   return error.toNonEmptyListOrNull()?.let { raise(it) } ?: results
 }
 
+/**
+ * Accumulate the errors obtained by executing the [transform] over every element of [NonEmptyList].
+ */
+@RaiseDSL
+public inline fun <Error, A, B> Raise<NonEmptyList<Error>>.mapOrAccumulate(
+  nonEmptyList: NonEmptyList<A>,
+  @BuilderInference transform: RaiseAccumulate<Error>.(A) -> B
+): NonEmptyList<B> {
+  val error = mutableListOf<Error>()
+  val results = ArrayList<B>(nonEmptyList.collectionSizeOrDefault(10))
+  for (item in nonEmptyList) {
+    fold<NonEmptyList<Error>, B, Unit>(
+      { transform(RaiseAccumulate(this), item) },
+      { errors -> error.addAll(errors) },
+      { results.add(it) }
+    )
+  }
+  return error.toNonEmptyListOrNull()?.let { raise(it) } ?: requireNotNull(results.toNonEmptyListOrNull())
+}
```

Comment:
To avoid duplicate code, we could also implement this function as:
```suggestion
): NonEmptyList<B> = requireNotNull(mapOrAccumulate(nonEmptyList as Iterable<A>, transform).toNonEmptyListOrNull())
```